### DISCUSSION TITLE
Combining multiple events into one

### DISCUSSION THREAD

**Discussion by Violet-Vibes** [wdelfuego/nova-calendar#27] [Q&A] [answered]
I'm using the calendar for an online fantasy textbased RPG, where tournaments are created in "series", with (by default) 10 minutes between each starting time. A serie has the same name and then appends the series number (in roman letters).

What I'm looking for is a way to combine/"merge" a serie of tournaments/events into one, showing the starting time of the first event and the last event.

Attached is a picture of how a couple of series currently looks like in the calendar, as you can see it gets messy very quickly if you have more than one tournament series running per day.

What I'm looking for is a way to customize the view so a series is merged down to one event, so for example on the 5th, the series "Oron gnager" would be combined into one event like this: 

```
Oron gnager I - IX
21:55 - 22:35
```

As for handling what happens when clicked, being taken to the first resource event would suffice or just having nothing happen, we use it as an overview to schedule tournaments without having them overlap with someone else's, as we have a few people with access to running/setting up tournaments.

![image](https://user-images.githubusercontent.com/19490536/181485792-284dca14-de4c-4c5e-bd57-00dd1128c4e9.png)

**Comment by wdelfuego** [comment]
A slightly hacky solution could be to use the `exclude` method to exclude all events but the first of every series, then in the `customizeEvent` method, for every first event that you do return, look up the last ending time for that series for that day and set the event end time to that. Then you'll have one event per series that spans the entire time range.

Are those events linked in the database in any way to reflect that they're from the same series, or are these all independent events in your database? It'd make sense to link them so you don't need to do name matching or something.

Another solution is to remove the resource from the `novaResources` method and return your own custom events using the `nonNovaEvents` method. Then you have to construct all the event objects yourself but you get complete control over how the data is presented in your calendar.

Edit: if you add an overarching TournamentSeries resource to link the specific events to, don't forget to give it a color property so you can easily distinguish your events on the calendar :)

**Reply by Violet-Vibes** [reply]
> A slightly hacky solution could be to use the `exclude` method to exclude all events but the first of every series, then in the `customizeEvent` method, for every first event that you do return, look up the last ending time for that series for that day and set the event end time to that. Then you'll have one event per series that spans the entire time range.
> 
> Are those events linked in the database in any way to reflect that they're from the same series, or are these all independent events in your database? It'd make sense to link them so you don't need to do name matching or something.
> 
> Another solution is to remove the resource from the `novaResources` method and return your own custom events using the `nonNovaEvents` method. Then you have to construct all the event objects yourself but you get complete control over how the data is presented in your calendar.
> 
> Edit: if you add an overarching TournamentSeries resource to link the specific events to, don't forget to give it a color property so you can easily distinguish your events on the calendar :)

Hm I'll look into it! The events are not linked in the database in anyway to reflect they are from the same series but now when you mentioned it, it does make sense to have something for it. Thank you :)

**Comment by wdelfuego** [comment, marked as answer]
As of version 1.6, you can implement this behavior using a [custom event generator](https://wdelfuego.github.io/nova-calendar/custom-event-generators.html). Simply implement it to return a single event for the first event of every series, set the end time of that event to the end time of the last of that series, and return no events for the rest of that series.
